### PR TITLE
Test server

### PR DESCRIPTION
**Pull request by rivalez**

### CODE REVIEW

File: battleship-server/src/main/java/com/academy/solid/nie/server/NetPlayer.java [hunk 1]
```diff
@@ -15,10 +15,14 @@
     private BufferedReader in;
 
     @Override
-    public void register(final ServerSocket serverSocket) throws IOException {
-        Socket clientSocket = serverSocket.accept();
-        out = new PrintWriter(new OutputStreamWriter(clientSocket.getOutputStream(), StandardCharsets.UTF_8), true);
-        in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream(), StandardCharsets.UTF_8));
+    public void register(final ServerSocket serverSocket) {
```
Comment (smolamarcin) [inline battleship-server/src/main/java/com/academy/solid/nie/server/NetPlayer.java]: final?

File: battleship-server/src/main/java/com/academy/solid/nie/server/NetPlayer.java [hunk 2]
```diff
@@ -15,10 +15,14 @@
...
+        try {
+            Socket clientSocket = serverSocket.accept();
+            out = new PrintWriter(new OutputStreamWriter(clientSocket.getOutputStream(), StandardCharsets.UTF_8), true);
+            in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream(), StandardCharsets.UTF_8));
+        } catch (IOException e) {
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/academy/solid/nie/server/NetPlayer.java]: this try-catch block make impossible to connect

File: battleship-server/src/main/java/com/academy/solid/nie/server/ShipSocketServer.java [hunk 3]
```diff
@@ -26,16 +25,12 @@
 
     @Override
     public void initialize() throws IOException {
-        ServerSocket serverSocket = new ServerSocket(PORT_NUMBER, 0, InetAddress.getByName(ip));
+        ServerSocket serverSocket = new ServerSocket(0, 0, InetAddress.getByName(ip));
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/academy/solid/nie/server/ShipSocketServer.java]: This should be the same as in Client and shouldn't be zero

File: battleship-server/src/main/java/com/academy/solid/nie/server/ShipSocketServer.java [hunk 4]
```diff
@@ -54,6 +49,19 @@ public void initialize() throws IOException {
         LOGGER.info("Initialized");
     }
 
+    List<Player> registerPlayers(ServerSocket serverSocket) {
+        List<Player> users = new ArrayList<>();
+        try {
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/academy/solid/nie/server/ShipSocketServer.java]: Are you sure this try-catch doesn't break flow

**Pull request closed without merging**